win32ss/user/ntuser/winsta.c

```diff
@@ -497,7 +497,20 @@ IntCreateWindowStation(
 
         InputWindowStation = WindowStation;
         WindowStation->Flags &= ~WSS_NOIO;
+
         InitCursorImpl();
+
+        UserCreateSystemThread(ST_DESKTOP_THREAD);
+        UserCreateSystemThread(ST_RIT);
+
+        /* Desktop functions require the desktop thread running so wait for it to initialize */
+        UserLeaveCo();
+        KeWaitForSingleObject( gpDesktopThreadStartedEvent,
```

Comment:
This can only be reached once, right? (it would hang forever the second time around due to the event being reset)

Reply:
Right